src/librustdoc/html/format.rs

```diff
@@ -281,61 +273,55 @@ pub(crate) fn print_where_clause<'a, 'tcx: 'a>(
-                            if f.alternate() {
-                                write!(f, "{ty_cx:#}: {generic_bounds:#}")
-                            } else {
-                                write!(f, "{ty_cx}: {generic_bounds}")
-                            }
-                        } else {
-                            if f.alternate() {
-                                write!(
-                                    f,
-                                    "for<{:#}> {ty_cx:#}: {generic_bounds:#}",
-                                    comma_sep(bound_params.iter().map(|lt| lt.print(cx)), true)
-                                )
-                            } else {
-                                write!(
+                    match pred {
+                        clean::WherePredicate::BoundPredicate { ty, bounds, bound_params } => {
+                            if !bound_params.is_empty() {
+                                f.write_str(if f.alternate() { "for<" } else { "for&lt;" })?;
+                                fmt::Display::fmt(
+                                    &comma_sep(bound_params.iter().map(|lt| lt.print(cx)), true),
                                     f,
-                                    "for&lt;{}&gt; {ty_cx}: {generic_bounds}",
-                                    comma_sep(bound_params.iter().map(|lt| lt.print(cx)), true)
-                                )
+                                )?;
+                                f.write_str(if f.alternate() { "> " } else { "&gt; " })?;
                             }
+                            fmt::Display::fmt(&ty.print(cx), f)?;
+                            f.write_str(":")?;
+                            if !bounds.is_empty() {
+                                f.write_str(" ")?;
+                                fmt::Display::fmt(&print_generic_bounds(bounds, cx), f)?;
+                            }
+                            Ok(())
                         }
-                    }
-                    clean::WherePredicate::RegionPredicate { lifetime, bounds } => {
-                        let mut bounds_display = String::new();
-                        for bound in bounds.iter().map(|b| b.print(cx)) {
-                            write!(bounds_display, "{bound} + ")?;
+                        clean::WherePredicate::RegionPredicate { lifetime, bounds } => {
+                            // We don't need to check `alternate` since we can be certain that neither
+                            // the lifetime nor the bounds contain any characters which need escaping.
+                            write!(f, "{}:", lifetime.print())?;
+                            if !bounds.is_empty() {
+                                write!(f, " {}", print_generic_bounds(bounds, cx))?;
+                            }
+                            Ok(())
                         }
-                        bounds_display.truncate(bounds_display.len() - " + ".len());
```

Comment:
This would overflow & wrap-around in release which didn't cause any issues but crash in debug builds of rustdoc. Just use the preexisting `print_generic_bounds`.